cmd/juju/storage/poolupdate.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package storage
+
+import (
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/utils/keyvalues"
+
+	jujucmd "github.com/juju/juju/cmd"
+	"github.com/juju/juju/cmd/modelcmd"
+)
+
+// PoolUpdateAPI defines the API methods that the storage commands use.
+type PoolUpdateAPI interface {
+	Close() error
+	UpdatePool(name string, attr map[string]interface{}) error
+	BestAPIVersion() int
+}
+
+const poolUpdateCommandDoc = `
+Update configuration attributes for a single existing storage pool.
+`
+
+// NewPoolUpdateCommand returns a command that updates the named storage pools' attributes.
+func NewPoolUpdateCommand() cmd.Command {
+	cmd := &poolUpdateCommand{}
+	cmd.newAPIFunc = func() (PoolUpdateAPI, error) {
+		return cmd.NewStorageAPI()
+	}
+	return modelcmd.Wrap(cmd)
+}
+
+// poolUpdateCommand updates a storage pool configuration attributes.
+type poolUpdateCommand struct {
+	PoolCommandBase
+	newAPIFunc  func() (PoolUpdateAPI, error)
+	poolName    string
+	configAttrs map[string]interface{}
+}
+
+// Init implements Command.Init.
+func (c *poolUpdateCommand) Init(args []string) (err error) {
+	if len(args) < 2 {
+		return errors.New("pool update requires name and configuration attributes")
```

Comment:
and type - we should allow for type also

Reply:
as per convo, type updates will be handled  via the passed config args.